test/unittests/execute_floating_point_test.cpp

```diff
@@ -71,6 +71,47 @@ TEST(execute_floating_point, f64_add)
     EXPECT_THAT(execute(*instance, 0, {1.0011, 6.0066}), Result(7.0077));
 }
 
+TEST(execute_floating_point, f64_promote_f32)
+{
+    /* wat2wasm
+    (func (param f32) (result f64)
+      local.get 0
+      f64.promote_f32
+    )
+    */
+    auto wasm = from_hex("0061736d0100000001060160017d017c030201000a070105002000bb0b");
+    auto instance = instantiate(parse(wasm));
+
+    using f32_limits = std::numeric_limits<float>;
+    using f64_limits = std::numeric_limits<double>;
+
+    EXPECT_THAT(execute(*instance, 0, {0.0f}), Result(0.0));
```

Comment:
Here you could do it as a table of cases + loop

Reply:
Done.